Sources/StreamChat/Controllers/ChannelListController/ChannelListController.swift

```diff
@@ -255,78 +260,8 @@ public class ChatChannelListController: DataController, DelegateCallable, DataSt
     }
 }
 
-/// When we receive events, we need to check if a channel should be added or removed from
-/// the current query depending on the following events:
-/// - Channel created: We analyse if the channel should be added to the current query.
-/// - New message sent: This means the channel will reorder and appear on first position,
-///   so we also analyse if it should be added to the current query.
-/// - Channel is updated: We only check if we should remove it from the current query.
-///   We don't try to add it to the current query to not mess with pagination.
 extension ChatChannelListController: EventsControllerDelegate {
-    public func eventsController(_ controller: EventsController, didReceiveEvent event: Event) {
-        if let channelAddedEvent = event as? NotificationAddedToChannelEvent {
-            linkChannelIfNeeded(channelAddedEvent.channel)
-        } else if let messageNewEvent = event as? MessageNewEvent {
-            linkChannelIfNeeded(messageNewEvent.channel)
-        } else if let messageNewEvent = event as? NotificationMessageNewEvent {
-            linkChannelIfNeeded(messageNewEvent.channel)
-        } else if let updatedChannelEvent = event as? ChannelUpdatedEvent {
-            unlinkChannelIfNeeded(updatedChannelEvent.channel)
-        } else if let channelVisibleEvent = event as? ChannelVisibleEvent, let channel = dataStore.channel(cid: channelVisibleEvent.cid) {
-            linkChannelIfNeeded(channel)
-        }
-    }
-
-    /// Handles if a channel should be linked to the current query or not.
-    private func linkChannelIfNeeded(_ channel: ChatChannel) {
-        guard !channels.contains(channel) else { return }
-        guard shouldChannelBelongToCurrentQuery(channel) else { return }
-        link(channel: channel)
-    }
-
-    /// Handles if a channel should be unlinked from the current query or not.
-    private func unlinkChannelIfNeeded(_ channel: ChatChannel) {
-        guard channels.contains(channel) else { return }
-        guard !shouldChannelBelongToCurrentQuery(channel) else { return }
-        worker.unlink(channel: channel, with: query)
-    }
-
-    /// Checks if the given channel should belong to the current query or not.
-    private func shouldChannelBelongToCurrentQuery(_ channel: ChatChannel) -> Bool {
-        if let filter = filter {
-            return filter(channel)
-        }
-
-        guard !client.config.isChannelAutomaticFilteringEnabled else {
-            // When auto-filtering is enabled the channel will appear or not automatically if the
-            // query matches the DB Predicate. So here we default to saying it always belong to the current query.
-            return true
-        }
-
-        // Fallback to legacy of checking if a channel should be linked to the current channel list query.
-        let deprecatedFallback: () -> Bool? = {
-            self.delegate?.controller(self, shouldAddNewChannelToList: channel)
```

Comment:
This is really old, and most likely nobody is using this anymore. If they are, they shouldn't really. Better to remove this function all together

Reply:
I will add an item about removing deprecated methods to v5.

Reply:
The thing is that if we are not using them already, not even has a fallback, then there's no point in waiting for v5 🤔

Reply:
Let me do another PR after this to delete some unused APIs.